src/python_testing/matter_testing_infrastructure/chip/testing/spec_parsing.py

```diff
@@ -518,50 +519,75 @@ class DataModelLevel(str, Enum):
     kDeviceType = 'device_types'
 
 
-def _get_data_model_root() -> str:
-    """Attempts to find ${CHIP_ROOT}/data_model or equivalent."""
+def _get_data_model_root() -> pathlib.PosixPath:
+    """Attempts to find the 'data_model' directory inside the 'chip.testing' package."""
+    try:
+        # Locate the directory where the 'chip.testing' module is located
+        package_dir = pathlib.Path(chip.testing.__file__).parent
 
-    # Since this class is generally in a module, we have to rely on being bootstrapped or
-    # we use CWD if we cannot
-    choices = [os.getcwd()]
+        # Construct the path to the 'data_model' directory (assuming this structure)
+        data_model_root = package_dir / 'data_model'
 
-    if 'PW_PROJECT_ROOT' in os.environ:
-        choices.insert(0, os.environ['PW_PROJECT_ROOT'])
+        if not data_model_root.exists():
+            raise FileNotFoundError(f"Data model directory not found in the package at {data_model_root}.")
 
-    for c in choices:
-        data_model_path = os.path.join(c, 'data_model')
-        if os.path.exists(os.path.join(data_model_path, 'master', 'scraper_version')):
-            return data_model_path
-    raise FileNotFoundError('Cannot find a CHIP_ROOT/data_model path. Tried %r as prefixes.' % choices)
+        return data_model_root
 
+    except Exception as e:
+        raise FileNotFoundError(f"Failed to find the data model root: {e}")
```

Comment:
is this except needed? I am concerned on nested handlers both here and below for `build_xml_clusters` because if you have:

```py
try:
   # ...
   raise X
catch Exception as e:
   raise Y(f"{e}")
```

then because X is an exception, the catch catches it so we are effectively throwing double the exceptions. Can we just remove this level of try/catch in both places?